websubhub-ballerina/content_distribution.bal

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/mime;
+
+# Processes the content publish request.
+# 
+# + caller - The `http:Caller` reference for the current request
+# + request - Received `http:Request` instance
+# + headers - The `http:Headers` received from the original `http:Request`
+# + params - Query parameters received in the original request
+# + adaptor - Current `websubhub:HttpToWebsubhubAdaptor`
+# + return - `error` if there is any error in execution or else `()`
+isolated function processPublishRequestAndRespond(http:Caller caller, http:Request request, http:Headers headers,
+                                                  map<string> params, HttpToWebsubhubAdaptor adaptor) returns error? {
+    http:Response response = new;
+    string? topic = getEncodedValueOrUpdatedErrorResponse(params, HUB_TOPIC, response);
+    if topic is () {
+        return error Error("Could not find the `hub.topic` parameter");
+    } else {
+        string contentType = request.getContentType();
+        UpdateMessage updateMsg = check createUpdateMessage(contentType, topic, request);
+        Acknowledgement|UpdateMessageError|error updateResult = adaptor.callOnUpdateMethod(updateMsg, headers);
```

Comment:
Just below is enough right ?

`Acknowledgement|error`

Reply:
* If there it is a `websubhub:UpdateMessageError` then we have to process the error details before responding. If it is an `error` then we use predefined response to respond.